lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
```

Comment:
my assumption was that `init/1` might not get called.

 This the way Phoenix does it
`def protect_from_forgery(conn, opts \\ []) do`
`Plug.CSRFProtection.call(conn, opts)`
`end` 

And the `csrf_protectection_test` is also written the same way (directly invoking `call/2`). please correct me if i'm missing something.